fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -4979,37 +4995,67 @@ public void dropTable(DropTableStmt stmt) throws DdlException {
                                     " please use \"DROP table FORCE\".");
                 }
             }
-            unprotectDropTable(db, table.getId(), stmt.isForceDrop(), false);
+            batchTaskMap = unprotectDropTable(db, table.getId(), stmt.isForceDrop(), false);
             DropInfo info = new DropInfo(db.getId(), table.getId(), -1L, stmt.isForceDrop());
             editLog.logDropTable(info);
         } finally {
             db.writeUnlock();
         }
-
+        sendDropTabletTasks(batchTaskMap);
         LOG.info("finished dropping table: {} from db: {}, is force: {}", tableName, dbName, stmt.isForceDrop());
     }
 
-    public boolean unprotectDropTable(Database db, long tableId, boolean isForceDrop, boolean isReplay) {
+    public void sendDropTabletTasks(HashMap<Long, AgentBatchTask> batchTaskMap) {
+        int numDropTaskPerBe = Config.max_agent_tasks_send_per_be;
+        for (Map.Entry<Long, AgentBatchTask> entry : batchTaskMap.entrySet()) {
+            AgentBatchTask originTasks = entry.getValue();
+            if (originTasks.getTaskNum() > numDropTaskPerBe) {
+                AgentBatchTask partTask = new AgentBatchTask();
+                List<AgentTask> allTasks = originTasks.getAllTasks();
+                int curTask = 1;
+                for (AgentTask task : allTasks) {
+                    partTask.addTask(task);
+                    if (curTask++ > numDropTaskPerBe) {
+                        AgentTaskExecutor.submit(partTask);
+                        curTask = 1;
+                        partTask = new AgentBatchTask();
+                        ThreadUtil.sleepAtLeastIgnoreInterrupts(1000);
+                    }
+                }
+                if (partTask.getAllTasks().size() > 0) {
+                    AgentTaskExecutor.submit(partTask);
+                }
+            } else {
+                AgentTaskExecutor.submit(originTasks);
+            }
+        }
+    }
+
+    public HashMap<Long, AgentBatchTask> unprotectDropTable(Database db, long tableId, boolean isForceDrop, boolean isReplay) {
+        HashMap<Long, AgentBatchTask> batchTaskMap = new HashMap<>();
         Table table = db.getTable(tableId);
         // delete from db meta
         if (table == null) {
-            return false;
+            return batchTaskMap;
         }
 
         table.onDrop();
 
         db.dropTable(table.getName());
         if (!isForceDrop) {
-            Catalog.getCurrentRecycleBin().recycleTable(db.getId(), table);
+            Table oldTable = Catalog.getCurrentRecycleBin().recycleTable(db.getId(), table);
+            if (oldTable != null && oldTable.getType() == TableType.OLAP) {
+                batchTaskMap = Catalog.getCurrentCatalog().onEraseOlapTable((OlapTable) oldTable, false);
```

Comment:
should not erase table here.

Reply:
this logic is move from recycleTable when eraseTableWithSameName remove oldTable send batch task.